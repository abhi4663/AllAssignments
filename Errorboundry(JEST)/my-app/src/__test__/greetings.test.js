/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-undef */
/* eslint-disable quotes */
import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Greetings from '../greetings';
test('test', () => {
  expect(true).toBe(true);
});

//after each test clean up the tests
afterAll(() => {
  cleanup();
});

//no yet testing weather component is testing or not

test('Should render greetings Component', () => {
  let User = {
    firstName: 'Abhishek',
    lastName: 'Patil',
  };
  render(<Greetings User={User} role='admin' />);
  const GreetingElement = screen.getByTestId('greetings-1');
  expect(GreetingElement).toBeInTheDocument();
  expect(GreetingElement).toHaveTextContent('Welcome Abhishek Patil');
});

test('matches snapshots', () => {
  let User = {
    firstName: 'Abhishek',
    lastName: 'Patil',
  };
  //creting a tree and then converting to JSON
  const tree = renderer.create(<Greetings User={User} role='admin' />).toJSON();
  //this is the component tree
  expect(tree).toMatchSnapshot(); //this creates a snapshot
  console.log(tree);
  //since snapshot alreday created so if any change in the component the snapshot will be updated
});
