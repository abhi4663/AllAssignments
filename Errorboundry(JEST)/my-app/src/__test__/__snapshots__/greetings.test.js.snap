// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshots 1`] = `
<div
  data-testid="greetings-1"
>
  Welcome 
  Abhishek
   
  Patil
</div>
`;
